src/node_id.rs

```diff
@@ -106,6 +104,34 @@ impl std::fmt::Debug for NodeId {
     }
 }
 
+/// Serialize with the 0x prefix.
+#[cfg(feature = "serde")]
+mod serde_hex_prfx {
+    pub fn serialize<T: AsRef<[u8]> + hex::ToHex, S: serde::Serializer>(
+        data: &T,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let dst = format!("0x{}", hex::encode(data));
+        serializer.serialize_str(&dst)
+    }
+
+    /// Deserialize with the 0x prefix.
+    pub fn deserialize<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+        T: hex::FromHex,
+        <T as hex::FromHex>::Error: std::fmt::Display,
+    {
+        let raw: &[u8] = serde::Deserialize::deserialize(deserializer)?;
+        let src = if raw.starts_with(b"0x") {
+            &raw[2..]
+        } else {
+            &raw[..]
+        };
```

Comment:
This changes the previous behaviour since now non-prefixed node_ids would be accepted. So I suggest to maintain the original behaviour and add this when/if there is the need

Reply:
the behaviour is similar to [`hex-serde`](https://github.com/fspmarshall/serde-hex/blob/master/src/lib.rs#L271C9-L275C11), do you suggest changing it?

Reply:
my suggestion is to keep the behaviour the same, if this is it then that's the way to go